src/crypto/hash_provider.h

```diff
@@ -88,6 +98,12 @@ namespace crypto
     {
       return ds::to_hex(h);
     };
+
+    static inline Sha256Hash from_string(const std::string& str)
```

Comment:
This may be prone to errors with the existing `Sha256Hash(const std::string& s)` constructor that expects an hex string. Maybe the existing constructor should be made like this function, as a `from_hex_string()` method?

Reply:
Yeah, it is, that's why it's split. I'll look for how many places need to change to remove the other ctor, but in general the digest API isn't super happy and deserves a cleanup pass.